src/recruiter-view/recruiter-view.service.ts

```diff
@@ -0,0 +1,203 @@
+      where: {
+        recruiterId: recruiterId,
+      },
+      include: [
+        {
+          model: SeasonModel,
+          as: "season",
+        },
+        {
+          model: CompanyModel,
+          as: "company",
+        },
+      ],
+    };
+
+    const pageOptions = parsePagesize(where);
+    Object.assign(findOptions, pageOptions);
+    parseFilter(findOptions, where.filterBy || {});
+    findOptions.order = parseOrder(where.orderBy || {});
+
+    const ans = await this.jobRepo.findAll(findOptions);
+
+    return ans.map((job) => job.get({ plain: true }));
+  }
+
+  async getJob(id: string, recruiterId: string) {
+    const ans = await this.jobRepo.findByPk(id, {
+      include: [
+        {
+          model: SeasonModel,
+          as: "season",
+        },
+        {
+          model: CompanyModel,
+          as: "company",
+        },
+        {
+          model: JobCoordinatorModel,
+          as: "jobCoordinators",
+          include: [
+            {
+              model: TpcMemberModel,
+              as: "tpcMember",
+              include: [
+                {
+                  model: UserModel,
+                  as: "user",
+                },
+              ],
+            },
+          ],
+        },
+        {
+          model: EventModel,
+          as: "events",
+          where: {
+            visibleToRecruiter: true,
+          },
+          required: false,
+        },
+        {
+          model: SalaryModel,
+          as: "salaries",
+        },
+      ],
+    });
+
+    if (!ans) throw new UnauthorizedException(`Unauthorized`);
+
+    if (ans.recruiterId !== recruiterId) {
+      throw new UnauthorizedException(`Unauthorized`);
+    }
+
+    return ans.get({ plain: true });
+  }
+
+  async getEvent(eventId: string, recruiterId: string) {
+    const ans = await this.eventRepo.findByPk(eventId, {
+      include: [
+        {
+          model: ApplicationModel,
+          as: "applications",
+          required: true,
+          include: [
+            {
+              model: ResumeModel,
+              as: "resume",
+              required: true,
+            },
+          ],
+        },
+        {
+          model: JobModel,
+          as: "job",
+        },
+      ],
+    });
+
+    if (!ans) throw new UnauthorizedException(`Unauthorized`);
+
+    if (ans.job.recruiterId !== recruiterId || ans.visibleToRecruiter === false) {
+      throw new UnauthorizedException(`Unauthorized`);
+    }
+
+    return ans.get({ plain: true });
+  }
+
+  async getEnums() {
+    return {
+      Domain: Object.values(IndustryDomainEnum),
+      Category: Object.values(CategoryEnum),
+    };
+  }
+
+  async updateRecruiter(recruiter: UpdateRecruiterDto, recruiterId: string, t: Transaction) {
+    const recruiterUpdatePromise = this.recruiterRepo.update(omit(recruiter, "user", "company"), {
+      where: { id: recruiterId },
+      transaction: t,
+    });
+
+    const userUpdatePromise = recruiter.user
```

Comment:
I am not sure we need a conditional statement. what if we pass an undefined in update? not sure on this, but we can see